### PR TITLE
[Angular] Add additionalDI config for add custom dependency injections

### PR DESCRIPTION
**Pull request by Justinidlerz**
Due to PR #4508 feature will overwrite the params for the constructor of the basic Query service to makes injected service lose.
Add the `additionalDI ` and config to add custom dependency injections
For the case see https://github.com/dotansimha/graphql-code-generator/issues/4366#issuecomment-759314958

### CODE REVIEW

**Comment by Justinidlerz** [comment]
Hello @dotansimha, I change to a new way to resolve the problem, could help to review and release a new version?

**Comment by dotansimha** [comment]
Thank you @Justinidlerz :) 
@kamilkisiela can you please review?

File: packages/plugins/typescript/apollo-angular/tests/apollo-angular.spec.ts [hunk 1]
```diff
@@ -107,6 +107,28 @@ describe('Apollo Angular', () => {
...
+          additionalDI: ['testService: TestService', 'testService1: TestService1'],
+        },
+        {
+          outputFile: 'graphql.ts',
+        }
+      )) as Types.ComplexPluginOutput;
+
+      expect(content.content).toBeSimilarStringTo(`
+          constructor(apollo: Apollo.Apollo, testService: TestService, testService1: TestService1) {
+            super(apollo, testService, testService1);
```
Comment (kamilkisiela) [inline packages/plugins/typescript/apollo-angular/tests/apollo-angular.spec.ts:125]: I'm not sure we need to pass those additional services to `super`

**Comment by kamilkisiela** [comment]
@Justinidlerz I left a comment, would you mind to explain why it's done this way?

File: packages/plugins/typescript/apollo-angular/tests/apollo-angular.spec.ts (hunk 1 above)
Comment (Justinidlerz) [reply packages/plugins/typescript/apollo-angular/tests/apollo-angular.spec.ts:125]: > @Justinidlerz I left a comment, would you mind to explain why it's done this way?

Yes, of course.

I'm using the `apolloAngularPackage` options to overwrite the basic angular-graphQL pkg and add some business logic in the basic service class. 

For example:

codegen.yml
```yml
      apolloAngularPackage: './custom.service'
```

custom.service.ts
```ts
import * as Apollo from 'apollo-angular';
import {Injectable} from '@angular/core';
import {
  EmptyObject,
  QueryOptionsAlone,
  WatchQueryOptionsAlone,
} from 'apollo-angular/types';
import {QueryRef} from 'apollo-angular';
import {Observable} from 'rxjs';
import {ApolloQueryResult} from '@apollo/client/core';
import {tap} from 'rxjs/operators';
import {NzMessageService} from 'ng-zorro-antd/message';
import {ScrollService} from '../scroll.service'
import {GoogleAnalysisService} from '../googleAnalysis.service'

export * from 'apollo-angular';

interface IWatchOptionExare<V> extends QueryOptionsAlone<V> {
   disabledScroll?: boolean;
   disabledToastError?: boolean;
}

@Injectable()
export class Query<T = {}, V = EmptyObject> extends Apollo.Query<T, V> {
  constructor(
    protected apollo: Apollo.Apollo,
    protected message: NzMessageService, // Add some other service in the base service class
    protected scroll: ScrollService,
    protected ga: GoogleAnalysisService,
  ) {
    super(apollo);
  }

  public watch(
    variables?: V,
    options?: WatchQueryOptionsAlone <V>
  ): QueryRef<T, V> {
    const watcher = super.watch(variables, options);

    watcher.valueChanges = watcher.valueChanges.pipe(
      tap(null, (err) => {
           this.message.error(err.message);
      })
    );

    return watcher;
  }

  public fetch(
    variables?: V,
    options?: IWatchOptionExare<V>
  ): Observable<ApolloQueryResult<T>> {

     if (!options.disabledScroll) { // Do some business logic before request
           this.scroll.toTop();
           this.ga.analysis()
           // etc...
     }
    return super.fetch(variables, options);
  }
}
```

Comment (kamilkisiela) [reply packages/plugins/typescript/apollo-angular/tests/apollo-angular.spec.ts:125]: okok, sounds good.

**Review by kamilkisiela — APPROVED** [review]

**Comment by kamilkisiela** [comment]
@dotansimha `changeset` and ready to go?

**Comment by dotansimha** [comment]
Thank you @Justinidlerz !

**Pull request merged**